lib/puppet_x/cisco/cmnutils.rb

```diff
@@ -112,6 +112,61 @@ def self.range_summarize(range_str)
       end
     end # class Utils
 
+    # PuppetX::Cisco::EncapUtil - Common Encapsulation methods used by
+    # Encapsulation Types/Providers
+    class EncapUtils
+      def self.batch_the_string(string)
+        EncapUtils.unsorted_list_to_string(EncapUtils.string_to_array(string.to_s))
+      end
+
+      # This will expand the string to a list of bds as integers
+      def self.string_to_array(string)
+        list = []
+        narray = string.split(',')
+        narray.each do |elem|
+          if elem.include?('-')
+            es = elem.gsub('-', '..')
+            ea = es.split('..').map { |d| Integer(d) }
+            er = ea[0]..ea[1]
+            list << er.to_a
+          else
+            list << elem.to_i
+          end
+        end
+        list.flatten
+      end
+
+      # This method will generate a batched string if a list is passed as
+      # argument
+      # Input would be as [1,2,3,4,5,10,11,12,7,30,100,31,32]
+      # output will be 1-5,10-12,7,30,100,31-32
```

Comment:
These methods seem to be attempting to solve the same problem that's handled by `range_summarize` (which probably needs a refactoring). What are you doing differently? Is there any way to solve these problems with a common utility?